src/TabBar.tsx

```diff
@@ -155,7 +178,41 @@ export default class TabBar<T extends Route> extends React.Component<
   };
 
   private getScrollAmount = (props: Props<T>, state: State, i: number) => {
-    const { layout } = state;
+    const { layout, tabWidths } = state;
+
+    if (props.dynamicWidth) {
+      const targetIndex =
+        i > props.navigationState.index &&
+        i < props.navigationState.routes.length - 1
+          ? Math.ceil(i)
+          : Math.floor(i);
+      let dynamicAmount = tabWidths
```

Comment:
Can you change the variable name to be more descriptive? I assume this means the scroll amount?